lib/domain/usecase/lookup_mxid_interactor.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:collection/collection.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/model/contact/contact.dart';
+import 'package:fluffychat/domain/model/contact/contact_status.dart';
+import 'package:fluffychat/domain/model/contact/lookup_list_mxid_request.dart';
+import 'package:fluffychat/domain/repository/lookup_repository.dart';
+import 'package:matrix/matrix.dart';
+
+class LookupMxidInteractor {
+  final LookupRepository _lookupRepository = getIt.get<LookupRepository>();
+
+  Stream<List<Contact>> execute({
+    required List<Contact> phonebookContacts,
+    int lookupChunkSize = 50,
+  }) async* {
+    try {
+      final hashDetails = await _lookupRepository.getHashDetails();
+      final thirdPartyIdToHashMap = {
+        for (final contact in phonebookContacts)
+          contact.thirdPartyId:
```

Comment:
why it is `thirdPartyId`? why not the `lookupAddress`?

Reply:
I mean: 
- `lookUpAddress: PhonebookContact`
- response `lookUpAddress:matrixId`

-> easy to find the PhonebookContact, then update with the matrixId?

Reply:
2 contacts maybe can have same email
so I will find email instead find contact